pkg/kubectl/cmd/run.go

```diff
@@ -149,64 +159,20 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 
 	params["env"] = cmdutil.GetFlagStringSlice(cmd, "env")
 
-	err = kubectl.ValidateParams(names, params)
-	if err != nil {
-		return err
-	}
-
-	obj, err := generator.Generate(params)
-	if err != nil {
-		return err
-	}
-
-	mapper, typer := f.Object()
-	version, kind, err := typer.ObjectVersionAndKind(obj)
-	if err != nil {
-		return err
-	}
-
-	inline := cmdutil.GetFlagString(cmd, "overrides")
-	if len(inline) > 0 {
-		obj, err = cmdutil.Merge(obj, inline, kind)
-		if err != nil {
+	if cmdutil.GetFlagBool(cmd, "expose") {
```

Comment:
The service creation will work, but the replication controller create will fail.  That seems reasonable to me, but I can add a pro-active check at the beginning that validates that the replication controller doesn't exist.

Of course even that's not safe against parallel CLI calls, we can't make the service and RC creation atomic...